ramlfications/parser/main.py

```diff
@@ -372,12 +372,70 @@ def create_resources(node, resources, root, parent):
             else:
                 child = _create_resource_node(name=k, raw_data=v, method=None,
                                               parent=parent, root=root)
-                resources.append(child)
+                if "type" in v:
+                    # There may be some more methods defined on the resource
+                    # type this one inherits from
+                    supertype = v["type"]
+                    resources = _create_supertype_resources(
+                        root, k, parent, supertype, resources)
+                else:
+                    resources.append(child)
             resources = create_resources(child.raw, resources, root, child)
 
     return resources
 
 
+def _create_supertype_resources(root, type_name, parent, supertype, resources):
```

Comment:
I've used `parent` (and `child`) wording for nested resources; I think I'd rather call this parent types (and same with the parameter name)